compiler/rustc_mir_dataflow/src/move_paths/builder.rs

```diff
@@ -447,8 +447,10 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
         match *operand {
             Operand::Constant(..) | Operand::Copy(..) => {} // not-a-move
             Operand::Move(place) => {
-                // a move
-                self.gather_move(place);
+                if !place.is_subtype() {
```

Comment:
this code should be removed

Reply:
If I remove that compiler won't compile because we are getting 

`No move errors should be allowed in MIR borrowck`

Reply:
then this is a bug with the lowering. Subtype projections are still ordinary moves.

Reply:
I moved this to new place, instead of skipping whole `Move` we only skip when we see projection `Subtype(T)` do we really need to record moves for `Subtype(T)`, they shouldn't affect outcome of the code.